### DISCUSSION TITLE
Trying to delete entire repeating segment but the tilde remains

### DISCUSSION THREAD

**Discussion by mayowan16** [nextgenhealthcare/connect#5786] [Q&A] [answered]
We are on Mirth Connect 4.0.1.  I am receiving DFT messages with 1 or 2 FT1.26 values.  I am trying to delete one specific value when it lands in either field and a specific CPT code lands in another segment.  The delete function seems to be working, but it is leaving the tilde behind even though I think I am deleting the entire segment.  Any suggestions?

You can see I tried deleting each individual field when the if statement requirements were met, but had the same problem with the tilde. 

Here is the code.
//DELETE TC modifier if found in any FT1 segment
if(msg['FT1']['FT1.26'][1].toString() > 0 &&	msg['FT1']['FT1.26'][1]['FT1.26.1'].toString() == 'TC' && msg['FT1']['FT1.7']['FT1.7.1'].toString() != '0055T')
	{
		delete msg['FT1']['FT1.26'][1]; //&& delete msg['FT1']['FT1.26'][1]['FT1.26.2'];
	}
	else if(msg['FT1']['FT1.26'][0].toString() >= 0 &&	msg['FT1']['FT1.26'][0]['FT1.26.1'].toString() == 'TC' && msg['FT1']['FT1.7']['FT1.7.1'].toString() != '0055T')
	{	
		delete msg['FT1']['FT1.26'][0]; //['FT1.26.1'] && delete msg['FT1']['FT1.26'][0]['FT1.26.2'];
	}
		else
	{
		;
	}

**Comment by jonbartels** [comment]
Can you post the XML representation of the message (the "transformed" view in the Mirth message viewer)?

When you post the XML please use code formatting so it is easier to read.

**Reply by pacmano1** [reply]
Paste a before and after message also please.

**Comment by tonygermano** [comment]
I don't see anything immediately wrong with your code, but it's hard to say without a test message as others have requested. This is how I would write what I think you're trying to do.

```js
for each (var ft1 in msg.FT1) { // for each FT1 segment
    if (ft1['FT1.7']['FT1.7.1'].toString() != '0055T') { // if the current FT1 segment does not have '0055T' in FT1-7.1
        for each (var ft1_26 in ft1['FT1.26']) { // for each FT1-26 in the current FT1 segment
            if (ft1_26['FT1.26.1'].toString() == 'TC' // if the current repetition of FT1-26 has TC in the first component
                delete ft1_26[0] // delete the current FT1-26
            }
        }
    }
}
```

**Reply by tonygermano** [reply]
Actually, I do see what is probably wrong. When you are checking the value in FT1-7, you aren't specifying which FT1 segment to pull it from.

**Reply by tonygermano** [reply]
Actually, you aren't accounting for multiple FT1 segments at all

**Comment by mayowan16** [comment]
Sorry about the formatting.  I am still very new to this.  

Here is the before for FT1.26
<FT1.26>
            <FT1.26.1>LT</FT1.26.1>
            <FT1.26.2>LEFT SIDE</FT1.26.2>
</FT1.26>
<FT1.26>
            <FT1.26.1>TC</FT1.26.1>
            <FT1.26.2>TECHNICAL COMPONENT</FT1.26.2>
</FT1.26>


Here is the after
        <FT1.26>
            <FT1.26.1>LT</FT1.26.1>
            <FT1.26.2>LEFT SIDE</FT1.26.2>
        </FT1.26>
        <FT1.26/>

I did try the code from tonygermano, and that does see to be a more direct way to accomplish this.  My first test appears to have processed correctly.  I will test it out and respond back if I have any other issues.  Thank you both!!!

**Reply by jonbartels** [reply, marked as answer]
No worries on the formatting.  This can help make code and errors easier to read: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

The after code block explains why you were seeing an extra `~`
```
<FT1.26>
<FT1.26.1>LT</FT1.26.1>
<FT1.26.2>LEFT SIDE</FT1.26.2>
</FT1.26>
<FT1.26/>
```

See how there are two elements `FT1.26` but one of them is a self-closing empty element? When Mirth serializes that XML to standard HL7 it would serialize as `FT1|...||LT^LEFT SIDE~||||||` . There are TWO `FT1.26` elements but one is empty, so the tilde has to be there to show the repeating field.
